.github/workflows/release.yml

```
@@ -249,3 +249,53 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           TAG: ${{ steps.create-release.outputs.computed-prefix }}${{ steps.create-release.outputs.version }}
+
+  check-plugin-version:
+    name: Check if plugin version exist in the registry and upload if not
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check version
+        id: check-version
+        run: |
+          set -eo pipefail
+
+          snforge_scarb_plugin_version=$(grep version crates/snforge-scarb-plugin/Scarb.toml | cut -d '"' -f 2)
+          snforge_scarb_plugin=$(curl -s https://scarbs.xyz/api/v1/index/sn/fo/snforge_scarb_plugin.json | jq --arg version $snforge_scarb_plugin_version '[.[] | select(.v == $version)] | length > 0')
+          echo "snforge_scarb_plugin=$snforge_scarb_plugin"
+          echo "snforge_scarb_plugin=$snforge_scarb_plugin" >> $GITHUB_OUTPUT
+
+      - name: Publish snforge_scarb_plugin
+        if: steps.check-version.outputs.snforge_scarb_plugin == 'false'
+        uses: ./.github/workflows/publish_plugin.yml
+        run: echo "Publishing snforge_scarb_plugin"
+
+  upload-to-registry:
+    name: Upload packages to the registry
+    runs-on: ubuntu-latest
+    needs: [ check-plugin-version, verify-version ]
```

```suggestion
    needs: [ create-release ]
```

How about this, I think then it will only run on successfully created release.

---

> How about this, I think then it will only run on successfully created release.

Yes, but we still need to ensure that latest plugin version is released before publishing the stds.  

https://github.com/foundry-rs/starknet-foundry/pull/2461#discussion_r1847852740